## microbial profile | diversity | network | and interactions

### abundance / profile
| Tool/workflows | Language | Lastest version | Release date | Journal | Pub Date | Paper title | Notes |
| -------------- | -------- | --------------- | ------------ | ------- | -------- | ----------- | ----- |
| [coverM](https://github.com/wwood/CoverM) | Rust | v0.7.0 | 2024.1 | _Bioinformatics_ | 2025.4 | [CoverM: read alignment statistics for metagenomics](https://doi.org/10.1093/bioinformatics/btaf147)
| [Fairy](https://github.com/bluenote-1577/fairy) | Rust | v0.5.5 | 2024.5 | _Microbiome_ | 2024.8 | [Fairy: fast approximate coverage for multi-sample metagenomic binning.](https://doi.org/10.1186/s40168-024-01861-6)
| [Tronko](https://github.com/lpipes/tronko) | C | NoRelese |||| [A rapid phylogeny-based method for accurate community profiling of large-scale metabarcoding datasets](https://doi.org/10.7554/eLife.85794)
| [MUSET](https://github.com/CamilaDuitama/muset) | C++ | v0.5.1 | 2024.12 | _Bioinformatics_ | 2025.2 | [MUSET: Set of utilities for constructing abundance unitig matrices from sequencing data](https://doi.org/10.1093/bioinformatics/btaf054)
| [SingleM Microbial Fraction(SMF)](https://github.com/EisenRa/2024_soil_dark_matter_reply) | Python ||| _ISME Commun_ | 2024.9 | [Quantifying microbial DNA in metagenomes improves microbial trait estimation.](https://doi.org/10.1093/ismeco/ycae111)
| [Lyrebird](https://wwood.github.io/singlem/Lyrebird) | Python ||| |||



### diversity
| Tool/workflows | Language | Lastest version | Release date | Journal | Pub Date | Paper title | Notes |
| -------------- | -------- | --------------- | ------------ | ------- | -------- | ----------- | ----- |
| [q2-kmerizer](https://github.com/bokulich-lab/q2-kmerizer) |||| _mSystems_ | 2025.2 | [Integrating sequence composition information into microbial diversity analyses with k-mer frequency counting](https://doi.org/10.1128/msystems.01550-24)

### network
| Tool/workflows | Language | Lastest version | Release date | Journal | Pub Date | Paper title | Notes |
| -------------- | -------- | --------------- | ------------ | ------- | -------- | ----------- | ----- |
| [SpeSpeNet](https://tbb.bio.uu.nl/SpeSpeNet) | Web | v1.0 || _ISME Commun_ | 2025.2 | [SpeSpeNet: an interactive and user-friendly tool to create and explore microbial correlation networks](https://doi.org/10.1093/ismeco/ycaf036)
| [LUPINE](https://github.com/SarithaKodikara/LUPINE) | R | NoReleaseTag || _Microbiome_ | 2025.3 | [Microbial network inference for longitudinal microbiome studies with LUPINE](https://doi.org/10.1186/s40168-025-02041-w) | [tutorial](https://mixomics.org/2024/05/lupine/)


### interactions / associations
| Tool/workflows | Language | Lastest version | Release date | Journal | Pub Date | Paper title | Notes |
| -------------- | -------- | --------------- | ------------ | ------- | -------- | ----------- | ----- |
| [PHLAME](https://github.com/quevan/phlame) | Python | NoRelease|| bioRxiv | 2025.2 | [Intraspecies associations from strain-rich metagenome samples](https://doi.org/10.1101/2025.02.07.636498)
||||| _Nat Microbiol_ | 2024.10 | [Prediction of strain level phageâ€“host interactions across the Escherichia genus using only genomic information](https://doi.org/10.1038/s41564-024-01832-5) | [code and files](https://github.com/mdmparis/coli_phage_interactions_2023)
||||| _Nat Commun_ | 2024.5 | [Prediction of Klebsiella phage-host specificity at the strain level](https://doi.org/10.1038/s41467-024-48675-6) | [code and files](https://github.com/dimiboeckaerts/PhageHostLearn)
| [iPHoP](https://bitbucket.org/srouxjgi/iphop/) || v1.3.3 || _PLoS Biol_ | 2023.4 | [iPHoP: An integrated machine learning framework to maximize host prediction for metagenome-derived viruses of archaea and bacteria](https://doi.org/10.1371/journal.pbio.3002083)
| [Anpan](https://github.com/biobakery/anpan) | R ||| bioRxiv | 2025.7 | [Quantifying Metagenomic Strain Associations from Microbiomes with Anpan](https://doi.org/10.1101/2025.01.06.631550)