## pangenome相关

| Tool/workflows | Language | Lastest version | Release date | Journal | Pub Date | Paper title | Notes |
| -------------- | -------- | --------------- | ------------ | ------- | -------- | ----------- | ----- |
| [PanGraph](https://github.com/neherlab/pangraph) | Julia/Python | v0.7.3 | 2023.11 |_Microb Genom_ |  2023.6 | [PanGraph: scalable bacterial pan-genome graph construction](https://doi.org/10.1099/mgen.0.001034)
| [PanGenie](https://github.com/eblerjana/pangenie) | C++ | v3.1.0 | 2024.3 | _Nat Genet_ | 2022.4 | [Pangenome-based genome inference](https://doi.org/10.1038/s41588-022-01043-w)
| [Panaroo](https://github.com/gtonkinhill/panaroo) | Python | v1.5.0 | 2024.4 | _Genome Biol_ | 2020.7 | [Producing polished prokaryotic pangenomes with the Panaroo pipeline](https://doi.org/10.1186/s13059-020-02090-4)
| [CELEBRIMBOR](https://github.com/bacpop/CELEBRIMBOR) | Python | v0.1.2 | 2024.7 | _Bioinformatics_ | 2024.9 | [CELEBRIMBOR: core and accessory genes from metagenomes](https://doi.org/10.1093/bioinformatics/btae542)
| [PangeBlocks](https://github.com/AlgoLab/pangeblocks) | snakemake ||| _BMC Bioinformatics_ | 2024.11 | [PangeBlocks: customized construction of pangenome graphs via maximal blocks](https://doi.org/10.1186/s12859-024-05958-5)
| [PGGB](https://github.com/pangenome/pggb) | shell | v0.7.2 | 2024.10 | _Nat Methods_ | 2024.10 | [Building pangenome graphs](https://doi.org/10.1038/s41592-024-02430-3)
| [GrAnnoT](https://forge.ird.fr/diade/dynadiv/grannot) | Python | v1.0.3 | 2024.8 | _bioRxiv_ | 2025.2 | [GrAnnoT, a tool for effecient and reliable annotation transfer through pangenome graph](https://doi.org/10.1101/2025.02.26.640337)
| [Pangene](https://github.com/lh3/pangene) | C/JS | v1.1(r2311) | 2024.2 | _Bioinformatics_ | 2024.7 | [Exploring gene content with pangene graphs](https://doi.org/10.1093/bioinformatics/btae456)
