## assembly

此列表收集了与组装相关的工具，包括短读长、长度长和混合组装。

### short-reads
| Tool/workflows | Language | Lastest version | Release date | Journal | Pub Date | Paper title | Notes |
| -------------- | -------- | --------------- | ------------ | ------- | -------- | ----------- | ----- |
| [SOAPdenovo2](https://github.com/aquaskyline/SOAPdenovo2) | C | r242 | 2020.10 | _GigaSci_ | 2012.12 | [SOAPdenovo2: an empirically improved memory-efficient short-read de novo assembler.](https://doi.org/10.1186/2047-217X-1-18)
| [idba/IDBA-UD](https://github.com/loneknightpy/idba) | C++ | v1.1.3 | 2016.7 | _Bioinformatics_ | 2012.4 | [IDBA-UD: a de novo assembler for single-cell and metagenomic sequencing data with highly uneven depth](https://doi.org/10.1093/bioinformatics/bts174)
| [SPAdes/metaSPAdes](https://github.com/ablab/spades) | C++ | v4.0.0 | 2024.6 | _Genome Res_ | 2017.5 | [metaSPAdes: a new versatile metagenomic assembler.](https://doi.org/10.1101/gr.213959.116)
| [megahit](https://github.com/voutcn/megahit) | C++ | v0.3.3-a | 2015.7 | _Bioinformatics_ | 2015.1 | [MEGAHIT: an ultra-fast single-node solution for large and complex metagenomics assembly via succinct de Bruijn graph.](https://doi.org/10.1093/bioinformatics/btv033)
| [megahit](https://github.com/voutcn/megahit) | C++ | v1.2.9 | 2019.10 | _Methods_ | 2016.6 | [MEGAHIT v1.0: A Fast and Scalable Metagenome Assembler driven by Advanced Methodologies and Community Practices](https://doi.org/10.1016/j.ymeth.2016.02.020)
| [PenguiN](https://github.com/soedinglab/plass) | C | v5-cf8933 | 2024.3 | bioRxiv | 2024.8 | [Strain-resolved de-novo metagenomic assembly of viral genomes and microbial 16S rRNAs.](https://doi.org/10.1101/2024.03.29.587318)
| [BBMerge](https://sourceforge.net/projects/bbmap/) | Bash | v39.10 | 2024.9 | _PLoS One_ | 2017.10 | [BBMerge - Accurate paired shotgun read merging via overlap.](https://doi.org/10.1371/journal.pone.0185056)
| [FLASH](https://github.com/ebiggers/flash) | C | v1.2.11 | 2018.3 | _Bioinformatics_ | 2011.9 | [FLASH: fast length adjustment of short reads to improve genome assemblies.](https://doi.org/10.1093/bioinformatics/btr507)
| [ABySS](https://github.com/bcgsc/abyss) | C++/C | v2.3.9 | 2024.9 | _Genome Res_ |  2017.2 | [ABySS 2.0: resource-efficient assembly of large genomes using a Bloom filter.](http://www.genome.org/cgi/doi/10.1101/gr.214346.116) |


### long-reads
| Tool/workflows | Language | Lastest version | Release date | Journal | Pub Date | Paper title | Notes |
| -------------- | -------- | --------------- | ------------ | ------- | -------- | ----------- | ----- |
| [MaSuRCA](https://github.com/alekseyzimin/masurca) | M4/Perl | v4.1.1 | 2024.2 | _Bioinformatics_ | 2013.8 | [The MaSuRCA genome assembler.](https://doi.org/10.1371/journal.pcbi.1009860)
| [metaFlye](https://github.com/fenderglass/Flye) | C/C++ | v2.9.5 | 2024.8 | _Nat Methods_ | 2020.10 | [metaFlye: scalable long-read metagenome assembly using repeat graphs.](https://doi.org/10.1038/s41592-020-00971-x)
| [Canu](https://github.com/marbl/canu) | C++/C | v2.3(v2.2) | 2024.12(2021.8) | _Genome Res_ | 2017.3 | [Canu: scalable and accurate long-read assembly via adaptive k-mer weighting and repeat separation](https://genome.cshlp.org/content/27/5/722)
| [FALCON](https://github.com/PacificBiosciences/FALCON) | Python | v0.3.0 | 2018.8 |
| [miniasm](https://github.com/lh3/miniasm) | TeX/C | v0.3 | 2018.7 | _Bioinformatics_ | 2016.3 | [Minimap and miniasm: fast mapping and de novo assembly for noisy long sequences.](https://doi.org/10.1093/bioinformatics/btw152)
| [wtdbg2](https://github.com/ruanjue/wtdbg2) | C | v2.5 | 2019.09 | _Nat Methods_ | 2019.12 | [Fast and accurate long-read assembly with wtdbg2.](https://doi.org/10.1038/s41592-019-0669-3)
| [Trycycler](https://github.com/rrwick/Trycycler) | Python | v0.5.5 | 2024.3 | _Genome Biol_ | 2021.9 | [Trycycler: consensus long-read assemblies for bacterial genomes.](https://doi.org/10.1186/s13059-021-02483-z)
| [NextDenovo](https://github.com/Nextomics/NextDenovo) | C | 2.5.2 | 2023.3 | _Genome Biol_ | 2024.4 | [NextDenovo: an efficient error correction and accurate assembly tool for noisy long reads.](https://doi.org/10.1186/s13059-024-03252-4)
| [Autocycler](https://github.com/rrwick/Autocycler) | Rust | v0.4.0 | 2025.4 | bioRxiv | 2025.5 | [Autocycler: long-read consensus assembly for bacterial genomes](https://doi.org/10.1101/2025.05.12.653612) | A tool for generating consensus long-read assemblies for bacterial genomes
| [nanoMDBG](https://github.com/GaetanBenoitDev/metaMDBG) | C++ | v1.1 | 2024.12 | bioRxiv | 2025.4 | [High-quality metagenome assembly from nanopore reads with nanoMDBG](https://doi.org/10.1101/2025.04.22.649928)
| [myloasm](https://github.com/bluenote-1577/myloasm/) | Rust | v0.1.0 | 2025.5 || [A new high-resolution long-read metagenome assembler for even noisy reads](https://myloasm-docs.github.io/) |
| 

### hybrid
| Tool/workflows | Language | Lastest version | Release date | Journal | Pub Date | Paper title | Notes |
| -------------- | -------- | --------------- | ------------ | ------- | -------- | ----------- | ----- |
| [idba/IDBA-Hybrid](https://github.com/loneknightpy/idba) | C++ | v1.1.3 | 2016.7 | _Bioinformatics_ | 2012.4 | [IDBA-UD: a de novo assembler for single-cell and metagenomic sequencing data with highly uneven depth](https://doi.org/10.1093/bioinformatics/bts174)
| [SPAdes/metaSPAdes](https://github.com/ablab/spades) | C++ | v4.0.0 | 2024.6 | _Genome Res_ | 2017.5 | [metaSPAdes: a new versatile metagenomic assembler](https://doi.org/10.1101/gr.213959.116)
| [Aviary](https://github.com/rhysnewell/aviary) | Python | v0.9.2 | 2024.9 ||| A hybrid assembly and MAG recovery pipeline (and more!)
| [OPERA-MS](https://github.com/CSB5/OPERA-MS) | C++/Python | v0.8.3 | 2020.6 | _Nat Biotechnol_ | 2019.7 | [Hybrid metagenomic assembly enables high-resolution analysis of resistance determinants and mobile elements in human microbiomes](https://doi.org/10.1038/s41587-019-0191-2)
| [Unicycler](https://github.com/rrwick/Unicycler) | C++ | v0.5.1 | 2024.7 | _PloS Comput Biol_ | 2017.6 | [Unicycler: resolving bacterial genome assemblies from short and long sequencing reads.](https://doi.org/10.1371/journal.pcbi.1005595)
| [MaSuRCA](https://github.com/alekseyzimin/masurca) | M4/Perl | v4.1.1 | 2024.2 | _Genome Res_ | 2017.5 | [Hybrid assembly of the large and highly repetitive genome of Aegilops tauschii, a progenitor of bread wheat, with the mega-reads algorithm](https://doi.org/10.1101/gr.213405.116)
| [HyLight](https://github.com/LuoGroup2023/HyLight) | C++/Python | v1.0.0 | 2024.9 | _Nat Commun_ | 2024.10 | [HyLight: Strain aware assembly of low coverage metagenomes](https://doi.org/10.1038/s41467-024-52907-0)

## PB HiFi
| Tool/workflows | Language | Lastest version | Release date | Journal | Pub Date | Paper title | Notes |
| -------------- | -------- | --------------- | ------------ | ------- | -------- | ----------- | ----- |
| [hifiasm-meta](https://github.com/xfengnefx/hifiasm-meta/) | C++ | v0.3.2 | 2024.9 | _Genome Biol_ | 2024.4 | [Evaluating and improving the representation of bacterial contents in long-read metagenome assemblies](https://doi.org/10.1186/s13059-024-03234-6)
| [metaMDBG](https://github.com/GaetanBenoitDev/metaMDBG) | C++ | v1.1 | 2024.12 | _Nat Biotechnol_ | 2024.1 | [High-quality metagenome assembly from long accurate reads with metaMDBG](https://doi.org/10.1038/s41587-023-01983-6)

### polish
| Tool/workflows | Language | Lastest version | Release date | Journal | Pub Date | Paper title | Notes |
| -------------- | -------- | --------------- | ------------ | ------- | -------- | ----------- | ----- |
| [FMLRC2](https://github.com/HudsonAlpha/fmlrc2) | Rust | v0.1.8 | 2022.7 | _Mol Bio Evo_ | 2023.3 | [Polishing de novo nanopore assemblies of bacteria and eukaryotes with FMLRC2](https://doi.org/10.1093/molbev/msad048)
| [pilon](https://github.com/broadinstitute/pilon) | Scala | v1.24 | 2021.1 | _Plos ONE_ | 2014.11 | [Pilon: An Integrated Tool for Comprehensive Microbial Variant Detection and Genome Assembly Improvement.](https://doi.org/10.1371/journal.pone.0112963)
| [NextPolish](https://github.com/Nextomics/NextPolish) | Rust | v1.4.1 | 2022.7 | _Bioinformatics_ | 2019.11 | [NextPolish: a fast and efficient genome polishing tool for long read assembly.](https://doi.org/10.1093/bioinformatics/btz891)
| [Polypolish](https://github.com/rrwick/Polypolish) | Rust | v0.6.0 | 2024.1 | _Microb Genom_ | 2024.6 | [How low can you go? Short-read polishing of Oxford Nanopore bacterial genome assemblies.](https://doi.org/10.1099/mgen.0.001254)
| [Pypolca](https://github.com/gbouras13/pypolca) | Python | v0.3.1 | 2024.2 | _Microb Genom_ | 2024.6 | [How low can you go? Short-read polishing of Oxford Nanopore bacterial genome assemblies.](https://doi.org/10.1099/mgen.0.001254)
| [MaSuRCA/POLCA](https://github.com/alekseyzimin/masurca) | M4/Perl | v4.1.1 | 2024.2 | _PLoS Comput Biol_ | 2020.6 | [The genome polishing tool POLCA makes fast and accurate corrections in genome assemblies.](https://doi.org/10.1371/journal.pcbi.1007981)
| [Racon](https://github.com/lbcb-sci/racon) | C++ | v1.5.0 | 2021.12 | _Genome Res_ | 2017.1 | [Ultrafast consensus module for raw de novo genome assembly of long uncorrected reads](http://www.genome.org/cgi/doi/10.1101/gr.214270.116)
| [medaka]()
| [dorado]() - 
| [DeepPolisher](https://github.com/google/deeppolisher) |||| bioRxiv | 2024.9 | [Highly accurate assembly polishing with DeepPolisher](https://doi.org/10.1101/2024.09.17.613505)


### genome size estimation
| Tool/workflows | Language | Lastest version | Release date | Journal | Pub Date | Paper title | Notes |
| -------------- | -------- | --------------- | ------------ | ------- | -------- | ----------- | ----- |
| [LRGE](https://github.com/mbhall88/lrge) | Rust | v0.1.3 | 2024.12 | bioRxiv | 2024.12 | [Genome size estimation from long read overlaps](https://doi.org/10.1101/2024.11.27.625777) | Pure

### other elements assmeble
| Tool/workflows | Language | Lastest version | Release date | Journal | Pub Date | Paper title | Notes |
| -------------- | -------- | --------------- | ------------ | ------- | -------- | ----------- | ----- |
| [Chopper](https://github.com/kschimke/Chopper) | Python | NoRelease || bioRxiv | 2025.1 | [Sequencing complete plasmids on Oxford Nanopore Technology Sequencers using R2C2 and Chopper](https://doi.org/10.1101/2025.01.16.633418)
| [PlasMAAG](https://github.com/RasmussenLab/vamb/tree/vamb_n2v_asy/workflow_PlasMAAG) | snakemake | part of VAMB || bioRxiv | 2025.3 | [Accurate plasmid reconstruction from metagenomics data using assembly-alignment graphs and contrastive learning](https://doi.org/10.1101/2025.02.26.640269)


### improvement
| Tool/workflows | Language | Lastest version | Release date | Journal | Pub Date | Paper title | Notes |
| -------------- | -------- | --------------- | ------------ | ------- | -------- | ----------- | ----- |
| [COBRA](https://github.com/linxingchen/cobra) | Python | v1.2.3 | 2024.2 | _Nat Microbiol_ | 2024.2 | [COBRA improves the completeness and contiguity of viral genomes assembled from metagenomes](https://doi.org/10.1038/s41564-023-01598-2)
