## bin

### tools and workflows
| Tool/workflows | Language | Lastest version | Release date | Journal | Pub Date | Paper title | Notes |
| -------------- | -------- | --------------- | -------------------- | ----------- | -------- | ----------- | ----- |
| [COCACOLA](https://github.com/younglululu/COCACOLA) | MATLAB/C++ || 2015 | _Bioinformatics_ | 2017.3 | [COCACOLA: binning metagenomic contigs using sequence COmposition, read CoverAge, CO-alignment and paired-end read LinkAge](https://doi.org/10.1093/bioinformatics/btw290) |
| [MetaBAT2](https://bitbucket.org/berkeleylab/metabat) | C++ | v2.17 | 2023.09 | _PeerJ_ | 2019.07 | [MetaBAT 2: an adaptive binning algorithm for robust and efficient genome reconstruction from metagenome assemblies](https://doi.org/10.7717/peerj.7359) |
| [MaxBin2](https://sourceforge.net/projects/maxbin2/) | C++ | 2.2.7 | 2020.06 | _Bioinformatics_ | 2015.10 | [MaxBin 2.0: an automated binning algorithm to recover genomes from multiple metagenomic datasets](https://doi.org/10.1093/bioinformatics/btv638) |
| [CONCOCT](https://github.com/BinPro/CONCOCT) | Python | v1.1.0 | 2019.08 | _Nat Methods_ | 2014.09 | [Binning metagenomic contigs by coverage and composition](https://doi.org/10.1038/nmeth.3103) |
| [VAMB](https://github.com/RasmussenLab/vamb) | Python | v4.1.3 | 2023.06 | _Nat Biotechnol_ | 2021.01 | [Improved metagenome binning and assembly using deep variational autoencoders](https://doi.org/10.1038/s41587-020-00777-4) |
| [TaxVAMB](https://github.com/RasmussenLab/vamb) | Python | v4.1.3 | 2023.6 | bioRxiv | 2024.10 | [Binning meets taxonomy: TaxVAMB improves metagenome binning using bi-modal variational autoencoder](https://doi.org/10.1101/2024.10.25.620172) |
| [UniteM](https://github.com/dparks1134/UniteM) | Python | 1.2.4 | 2022.10 |
| [SemiBin](https://github.com/BigDataBiology/SemiBin) | Python | v1.5.1 | 2023.03 | _Nat Commun_ | 2022.04 | [A deep siamese neural network improves metagenome-assembled genomes in microbiome datasets across different environments](https://doi.org/10.1038/s41467-022-29843-y) |
| [SemiBin2](https://github.com/BigDataBiology/SemiBin) | Python | v2.1.0 | 2024.03 | _Bioinformatics_ | 2023.06 | [SemiBin2: self-supervised contrastive learning leads to better MAGs for short- and long-read sequencing](https://doi.org/10.1093/bioinformatics/btad209) | [Benchmark](https://github.com/BigDataBiology/SemiBin2_benchmark) |
| [Smeta](https://github.com/YuhaoZhangwow/SMeta) | C++ | NoReleaseTag | 2024.09 | bioRxiv | 2024.08 | [SMeta, a binning tool using single-cell sequences to aid in reconstructing species from metagenome accurately.](https://doi.org/10.1101/2024.08.25.609542)
| [Bin Chiken](https://github.com/AroneyS/binchicken) | Python | v0.12.6 | 2024.12 | bioRxiv | 2024.11 | [Bin Chicken: targeted metagenomic coassembly for the efficient recovery of novel genomes](https://doi.org/10.1101/2024.11.24.625082) | 
| [MetaCoAG](https://github.com/metagentools/MetaCoAG) | Python | v1.2.2 | 2024.9 | _RECOMB_ | 2022.4 | [MetaCoAG: Binning Metagenomic Contigs via Composition, Coverage and Assembly Graphs](https://doi.org/10.1007/978-3-031-04749-7_5) |
| [GraphBin](https://github.com/metagentools/GraphBin) | Python | v1.7.4 | 2024.8 | _Bioinformatics_ | 2020.6| [GraphBin: Refined binning of metagenomic contigs using assembly graphs](https://doi.org/10.1093/bioinformatics/btaa180) |
| [GraphBin2](https://github.com/metagentools/GraphBin2) | Python | v1.3.3 | 2024.9 | _Algorithms Mol Biol_ | 2021.5 | [Improving metagenomic binning results with overlapped bins using assembly graphs](https://doi.org/10.1186/s13015-021-00185-6) |
| [MetaBinner](https://github.com/ziyewang/MetaBinner) | Python/Perl | v1.4.4 | 2022.9 | _Genome Biol_ | 2023.1 | [MetaBinner: a high-performance and stand-alone ensemble binning method to recover individual genomes from complex microbial communities](https://doi.org/10.1186/s13059-022-02832-6) | 
| [[galah](https://github.com/wwood/galah)] | Rust | v0.4.2 | 2024.09 ||| More scalable dereplication for metagenome assembled genomes |

### MAG improvement
| Tool/workflows | Language | Lastest version | Release date | Journal | Pub Date | Paper title | Notes |
| -------------- | -------- | --------------- | ------------ | ------- | -------- | ----------- | ----- |
| [Circlator](https://github.com/sanger-pathogens/circlator) | Python | v1.5.5 | 2020.10 | _Genome Biol_ | 2015.12 | [Circlator: automated circularization of genome assemblies using long sequencing reads](https://doi.org/10.1186/s13059-015-0849-0) |
| [gapFinisher](https://github.com/kammoji/gapFinisher) ||| 2019 | _PloS One_ | 2019.9 | [gapFinisher: A reliable gap filling pipeline for SSPACE-LongRead scaffolder output](https://doi.org/10.1371/journal.pone.0216885) |
| [FGAP](https://github.com/pirovc/fgap) | MATLAB | v1.8.1 | 2017.11 | _BMC Res Notes_ | 2014.6 | [FGAP: an automated gap closing tool](http://doi.org/10.1186/1756-0500-7-371) |
| [MaSuRCA/samba.sh](https://github.com/alekseyzimin/masurca) |||| _PLoS Comput Biol_ | 2022.2 | [The SAMBA tool uses long reads to improve the contiguity of genome assemblies.](https://doi.org/10.1371/journal.pcbi.1009860) |

### strain-level resolve
| Tool/workflows | Language | Lastest version | Release date | Journal | Pub Date | Paper title | Notes |
| -------------- | -------- | --------------- | ------------ | ------- | -------- | ----------- | ----- |
| [inStrain](https://github.com/MrOlm/inStrain) | Python | v1.9.0 | 2024.5 | _Nat Biotechnol_ | 2021.1 | [inStrain profiles population microdiversity from metagenomic data and sensitively detects shared microbial strains](https://doi.org/10.1038/s41587-020-00797-0) |
| [Floria](https://github.com/bluenote-1577/floria) | Rust | v0.0.1 | 2024.1 | _Bioinformatics_ | 2024.6 | [Floria: fast and accurate strain haplotyping in metagenomes](https://doi.org/10.1093/bioinformatics/btae252) |
| [Lorikeet](https://github.com/rhysnewell/Lorikeet) | Rust | v0.8.2 | 2023.12 ||| Strain resolver for metagenomics |
| [STRONG](https://github.com/chrisquince/STRONG) | Python | v0.0.1-beta | 2021.5 | _Genome Biol_ | 2021.7 | [STRONG: metagenomics strain resolution on assembly graphs](https://doi.org/10.1186/s13059-021-02419-7) |
| [Strainberry](https://github.com/rvicedomini/strainberry) | Python | v1.1 | 2021.5 | _Nat Commun_ | 2021.7 | [Strainberry: automated strain separation in low-complexity metagenomes using long reads](https://doi.org/10.1038/s41467-021-24515-9) |
| [inStrain](https://github.com/MrOlm/inStrain) | Python | v1.3.1 | 2020.8 | _Nat Biotechnol_ | 2021.1 | [inStrain profiles population microdiversity from metagenomic data and sensitively detects shared microbial strains](https://doi.org/10.1038/s41587-020-00797-0) |
| [Strainy](https://github.com/katerinakazantseva/strainy) | Python | v1.1 | 2024.9 | _Nat Methods_ | 2024.9 | [Strainy: phasing and assembly of strain haplotypes from long-read metagenome sequencing](https://doi.org/10.1038/s41592-024-02424-1) |
| [VStrains](https://github.com/metagentools/VStrains) | Python | v1.1.0 | 2023.2 | bioRxiv | 2023.1 | [VStrains: De Novo Reconstruction of Viral Strains via Iterative Path Extraction From Assembly Graphs](https://doi.org/10.1101/2022.10.21.513181) |
| [devider](https://github.com/bluenote-1577/devider) | Rust | v0.0.1 | 2024.10 | bioRxiv | 2024.11 | [devider: long-read reconstruction of many diverse haplotypes](https://doi.org/10.1101/2024.11.05.621838) |
| [ChronoStrain](https://github.com/gibsonlab/chronostrain) | Python | v0.6.0 | 2025.4 | _Nat Microbiol_ | 2025.3 | [Longitudinal profiling of low-abundance strains in microbiomes with ChronoStrain](https://doi.org/10.1038/s41564-025-01983-z)


### MAG assessment
| Tool/workflows | Language | Lastest version | Release date | Journal | Pub Date | Paper title | Notes |
| -------------- | -------- | --------------- | ------------ | ------- | -------- | ----------- | ----- |
[BUSCO](https://busco.ezlab.org) | Python | v5.8.1 | 2024.10 | _Mol Biol Evol_ | 2021.10 | [BUSCO update: novel and streamlined workflows along with broader and deeper phylogenetic coverage for scoring of eukaryotic, prokaryotic, and viral genomes](http://arxiv.org/abs/2106.11799) |
| [CheckM](https://github.com/Ecogenomics/CheckM) | Python | v1.2.3 | 2024.06 | _Genome Res_ |2015.07 | [CheckM: assessing the quality of microbial genomes recovered from isolates, single cells, and metagenomes](https://doi.org/10.1101/gr.186072.114) |
| [CheckM2](https://github.com/chklovski/CheckM2) | Python | v1.0.2 | 2023.05 | _Nat Methods_ | 2023.07 | [CheckM2: a rapid, scalable and accurate tool for assessing microbial genome quality using machine learning](https://doi.org/10.1038/s41592-023-01940-w) |
| [RefineM](https://github.com/dparks1134/RefineM) | Python | v0.1.2 | 2020.11 | _Nat Microbiol_ | 2017.09 | [Recovery of nearly 8,000 metagenome-assembled genomes substantially expands the tree of life](https://doi.org/10.1038/s41564-017-0012-7) | Unsupported
| [MAGpurify](https://github.com/snayfach/MAGpurify) | Python | v2.1.2 | 2020.03 | _Nature_ | 2019.03 | [New insights from uncultivated genomes of the global human gut microbiome](https://doi.org/10.1038/s41586-019-1058-x) |
| [GUNC](https://github.com/grp-bork/gunc) | Python | v1.0.6 | 2023.11 | _Genome Biol_ | 2021.06 | [GUNC: Detection of Chimerism and Contamination in Prokaryotic Genomes](https://doi.org/10.1186/s13059-021-02393-0) |
| [DFAST_QC](https://github.com/nigyta/dfast_qc) | Python | v1.0.5 | 2024.09 | bioRxiv | 2024.07 | [DFAST_QC: Quality Assessment and Taxonomic Identification Tool for Prokaryotic Genomes](https://doi.org/10.1101/2024.07.22.604526) |
| [dRep](https://github.com/MrOlm/drep) | Python | v3.4.2 | 2023.02 | _ISME J_ | 2017.07 | [dRep: a tool for fast and accurate genomic comparisons that enables improved genome recovery from metagenomes through de-replication](https://doi.org/10.1038/ismej.2017.126) |
| [DAS Tool](https://github.com/cmks/DAS_Tool) | R/Ruby | v1.1.7 | 2024.01 | _Nat Microbial_ | 2018.05 | [Recovery of genomes from metagenomes via a dereplication, aggregation and scoring strategy](https://doi.org/10.1038/s41564-018-0171-1) |
| [Binette](https://github.com/genotoul-bioinfo/Binette) | Python | v1.0.3 | 2024.9 ||| A fast and accurate binning refinement tool to constructs high quality MAGs from the output of multiple binning tools |
| [MAGqual](https://github.com/ac1513/MAGqual) | Python | v0.3.0 | 2024.8 | _Microbiome_ | 2024.11 | [MAGqual: a stand-alone pipeline to assess the quality of metagenome-assembled genomes](https://doi.org/10.1186/s40168-024-01949-z) |


### other genomes recovery
| Tool/workflows | Language | Lastest version | Release date | Journal | Pub Date | Paper title | Notes |
| -------------- | -------- | --------------- | ------------ | ------- | -------- | ----------- | ----- |
| [BEREN](https://gitlab.com/benminch1/BEREN) | Python | NoReleaseYet || bioRxiv | 2024.10 | [BEREN: A bioinformatic tool for recovering Giant viruses, Polinton-like Viruses, and Virophages in metagenomic data](https://doi.org/10.1101/2024.10.09.617401) |



