## annotation

### tools
| Tool/workflows | Language | Lastest version | Release date | Journal | Pub Date | Paper title | Notes |
| -------------- | -------- | --------------- | ------------ | ------- | -------- | ----------- | ----- |
| [eggNOG-mapper v2](https://github.com/eggnogdb/eggnog-mapper) | Python | v2.1.12 | 2023.8 | _Mol Biol Evol_ | 2021.10 | [eggNOG-mapper v2: functional annotation, orthology assignments, and domain prediction at the metagenomic scale](https://doi.org/10.1093/molbev/msab293)
| [KofamScan](https://www.genome.jp/ftp/tools/kofam_scan/) | Ruby | v1.3.0 | 2020.05 | _Bioinformatics_ | 2019.11 | [KofamKOALA: KEGG ortholog assignment based on profile HMM and adaptive score threshold](https://doi.org/10.1093/bioinformatics/btz859)
| [Macrel](https://github.com/BigDataBiology/macrel) | Python | v1.5.0 | 2024.9 | _PeerJ_ | 2020.12 | [Macrel: antimicrobial peptide screening in genomes and metagenomes](https://doi.org/10.7717/peerj.10555)
| [RGI](https://github.com/arpcard/rgi) | Python | 6.0.3 | 2023.9 | _Nucleic Acids Res_ | 2023.1 | [CARD 2023: expanded curation, support for machine learning, and resistome prediction at the Comprehensive Antibiotic Resistance Database](https://doi.org/10.1093/nar/gkac920)
| [MetaCerberus](https://github.com/raw-lab/metacerberus) | Python | v1.4.0 | 2024.8 | _Bioinformatics_ | 2024.2 | [MetaCerberus: distributed highly parallelized HMM-based processing for robust functional annotation across the tree of life](https://doi.org/10.1093/bioinformatics/btae119)
| [GMSC-mapper](https://github.com/BigDataBiology/GMSC-mapper) | Python | v0.1.0 | 2024.04 | _Nat Commun_ | 2024.08 | [A catalog of small proteins from the global microbiome](https://doi.org/10.1038/s41467-024-51894-6)
| [Bakta](https://github.com/oschwengers/bakta) | Python | v1.10.1 | 2024.11 | _Microb Genomics_ |  2021.11 | [Bakta: rapid and standardized annotation of bacterial genomes via alignment-free sequence identification](https://doi.org/10.1099/mgen.0.000685) | [web version](https://bakta.computational.bio)
| [DRAM](https://github.com/WrightonLabCSU/DRAM) | Python | v1.5.0 | 2024.1 | _Nucleic Acids Res_ | 2020.9 | [DRAM for distilling microbial metabolism to automate the curation of microbiome function](https://doi.org/10.1093/nar/gkaa621)
| [graftM](https://github.com/geronimp/graftM) | Python | v0.14.0 | 2022.5 | _Nucleic Acids Res_ | 2018.6 | [GraftM: a tool for scalable, phylogenetically informed classification of genes within metagenomes](https://doi.org/10.1093/nar/gky174)
| [CD-Search](http://www.ncbi.nlm.nih.gov/Structure/cdd/wrpsb.cgi) | Web | | | _Nucleic Acids Res_ | 2004.7 | [CD-Search: protein domain annotations on the fly](https://doi.org/10.1093/nar/gkh454)
| [HMMER3](https://github.com/EddyRivasLab/hmmer) | C | v3.4 | 2023.8 | _PLoS Comput Biol_ | 2011.10 | [Accelerated profile HMM searches](https://doi.org/10.1371/journal.pcbi.1002195)
| [DeepARG](https://github.com/gaarangoa/deeparg) | Python2.7 | | 2023.11 | _Microbiome_ | 2018.2 | [DeepARG: a deep learning approach for predicting antibiotic resistance genes from metagenomic data](https://doi.org/10.1186/s40168-018-0401-z)
| [ARGs-OAP](https://github.com/xinehc/args_oap) | Python | v3.2.4 | 2023.10(2023.6) | _Bioinformatics_ | 2016.8 | [ARGs-OAP: online analysis pipeline for antibiotic resistance genes detection from metagenomic data using an integrated structured ARG-database](https://doi.org/10.1093/bioinformatics/btw136)
| [ARGs-OAP3.0](https://github.com/xinehc/args_oap) | Python | v3.2.4 | 2023.10(2023.6) | _Engineering_ | 2023.8 | [ARGs-OAP v3.0: Antibiotic-Resistance Gene Database Curation and Analysis Pipeline Optimization.](https://doi.org/10.1016/j.eng.2022.10.011)
| [argNorm](https://github.com/BigDataBiology/argNorm) | Python | v0.7.0 | 2025.3(2024.8) | _Bioinformatics_ | 2025.4 | [argNorm: Normalization of antibiotic resistance gene annotations to the Antibiotic Resistance Ontology (ARO)](https://doi.org/10.1093/bioinformatics/btaf173)
| [OrthoLoger](https://orthologer.ezlab.org) | bash | v3.5.0 | 2024.10 | _Nucleic Acids Res_ | 2022.11 | [OrthoDB v11: annotation of orthologs in the widest sampling of organismal diversity](https://doi.org/10.1093/nar/gkac998)
| [nail](https://github.com/TravisWheelerLab/nail) | Rust | v0.2.0 | 2024.7 | bioRxiv | 2024.1 | [nail: software for high-speed, high-sensitivity protein sequence annotation](https://doi.org/10.1101/2024.01.27.577580)
| [FastOMA](https://github.com/DessimozLab/FastOMA/) | Python | v0.3.4 | 2024.9 | _Nat Methods_ | 2025.1 | [Orthology inference at scale with FastOMA](https://doi.org/10.1038/s41592-024-02552-8)
| [BenchAMRking](https://erasmusmc-bioinformatics.github.io/benchAMRking/) | web workflow AAMR detection ||| _BMC Genomics_ | 2025.1 | [BenchAMRking: a Galaxy-based platform for illustrating the major issues associated with current antimicrobial resistance (AMR) gene prediction workflows.](https://doi.org/10.1186/s12864-024-11158-5)
| [OMArk](https://github.com/DessimozLab/OMArk) | Python | v0.3.0 | 2023.10 | _Nat Biotechnol_ |  2024.2 | [Quality assessment of gene repertoire annotations with OMArk](https://doi.org/10.1038/s41587-024-02147-w)
| [mettannotator](https://github.com/EBI-Metagenomics/mettannotator) | Nextflow | 2024.12 | v1.4.0 | _Bioinformatics_ | 2025.1 | [mettannotator: a comprehensive and scalable Nextflow annotation pipeline for prokaryotic assemblies](https://doi.org/10.1093/bioinformatics/btaf037)
| [pseudofinder](https://github.com/filip-husnik/pseudofinder) | Python | v1.1.0 | 2022.3(2021.4) | _Mol Biol Evol_ | 2022.7 | [Pseudofinder: Detection of Pseudogenes in Prokaryotic Genomes](https://doi.org/10.1093/molbev/msac153)
| [AntiDeffenseFinder](https://defensefinder.mdmlab.fr/) | Web Service ||| _Nucleic Acids Res_ | 2025.1 | [Exploring the diversity of anti-defense systems across prokaryotes, phages and mobile genetic elements](https://doi.org/10.1093/nar/gkae1171)
| [Genomic conttext](https://github.com/bio-ontology-research-group/Genomic_context) | Python ||| bioRxiv | 2024.10 | [Context-based protein function prediction in bacterial genomes.](https://doi.org/10.1101/2024.10.14.618363) | a step by step tutorial
| [OMAnnotator](https://github.com/DessimozLab/OMAnnotator) | Python ||| bioRxiv | 2024.11 | [OMAnnotator: a novel approach to building an annotated consensus genome sequence](https://doi.org/10.1101/2024.12.04.626846)
| [MHCScan](https://github.com/Arkadiy-Garber/MHCscan) | Python ||| _Front Microbiol_ | 2024.10 | [Large-scale prediction of outer-membrane multiheme cytochromes uncovers hidden diversity of electroactive bacteria and underlying pathways](https://doi.org/10.3389/fmicb.2024.1448685)
| [DRAMMA](https://github.com/burstein-lab/DRAMMA) | Python ||| _Microbiome_ | 2025.3 | [DRAMMA: a multifaceted machine learning approach for novel antimicrobial resistance gene detection in metagenomic data](https://doi.org/10.1186/s40168-025-02055-4)
| [ChroQueTas](https://github.com/nmquijada/ChroQueTas) | Shell | v0.4.2 | 2024.10 | bioRxiv | 2024.10 | [FungAMR: A comprehensive portrait of antimicrobial resistance mutations in fungi](https://www.biorxiv.org/content/10.1101/2024.10.07.617009v1) | Fungi, work with FungAMR
| [MICROPHERRET](https://github.com/BizzoTL/MICROPHERRET/) | Python ||| _Environ Microbiome_ | 2024.8 | [MICROPHERRET: MICRObial PHEnotypic tRait ClassifieR using Machine lEarning Techniques](https://doi.org/10.1186/s40793-024-00600-6)


### databases
| Tool/workflows | Type | Lastest version | Release date | Journal | Pub Date | Paper title | Notes |
| -------------- | ---- | --------------- | ------------ | ------- | -------- | ----------- | ----- |
| [eggNOG 6.0](http://eggnog6.embl.de/) | AA Genes | v6.0 | 2022.09 | _Nucleic Acids Res_ | 2022.11 | [eggNOG 6.0: Enabling comparative genomics across 12,535 organisms](https://doi.org/10.1093/nar/gkac1022)
| [KEGG](https://www.genome.jp/kegg/) | AA Genes | 111.0 | 2024.08 | _Nucleic Acids Res_ | 2016.1 | [KEGG as a reference resource for gene and protein annotation](https://doi.org/10.1093/nar/gkv1070)
| [CAZy](http://www.cazy.org) | AA | web || _Nucleic Acids Res_ | 2022.1 | [The carbohydrate-active enzyme database: functions and literature](https://doi.org/10.1093/nar/gkab1045)
| [CARD](https://card.mcmaster.ca) | ARG | v3.3.0 | 2024.08 | _Nucleic Acids Res_ | 2023.01 | [CARD 2023: expanded curation, support for machine learning, and resistome prediction at the Comprehensive Antibiotic Resistance Database](https://doi.org/10.1093/nar/gkac920)
| [UniProt](https://www.uniprot.org) | AA genes | v2024_04 | 2024.07 | _Nucleic Acids Res_ | 2022.11 | [The UniProt Consortium. UniProt: the Universal Protein Knowledgebase in 2023](https://doi.org/10.1093/nar/gkac1052)
| [UniProt](https://www.uniprot.org/) | AA genes | v2024_5 || _Nucleic Acids Res_ | 2024.11 | [UniProt: the Universal Protein Knowledgebase in 2025.](https://doi.org/10.1093/nar/gkae1010)
| [ISOSDB](https://github.com/joshuakirsch/pseudoR) | IS Sequences | v3 | 2023.12 | _Cell Host Microbe_ | 2024.05 | [A metagenomics pipeline reveals insertion sequence-driven evolution of the microbiota](https://doi.org/10.1016/j.chom.2024.03.005)
| [AMPSphere](https://ampsphere.big-data-biology.org/home) | AMP | v2022-03 | 2022.03 | _Cell_ | 2024.07 | [Discovery of antimicrobial peptides in the global microbiome with machine learning](https://doi.org/10.1016/j.cell.2024.05.013)
| [DRAMP](http://dramp.cpu-bioinfor.org) | AMP | v4.0 | 2024.09 | _Nucleic Acids Res_ | 2022.01 | [DRAMP 3.0: an enhanced comprehensive data repository of antimicrobial peptides](https://doi.org/10.1093/nar/gkab651)
| [GMSC](https://gmsc.big-data-biology.org) | smORF | v1.0 | 2024.08 | _Nat Commun_ | 2024.08 | [A catalog of small proteins from the global microbiome](https://doi.org/10.1038/s41467-024-51894-6)
| [TCDB](https://www.tcdb.org) | transporters | 2024-09 | 2024.09 | _Nucleic Acids Res_ | 2021.01 | [The Transporter Classification Database (TCDB): 2021 update](https://doi.org/10.1093/nar/gkaa1004)
| [VFDB](http://www.mgc.ac.cn/VFs/main.htm) | virulence factors | 2024.9 || _Nucleic Acids Res_ | 2022.1 | [VFDB 2022: a general classification scheme for bacterial virulence factors](https://doi.org/10.1093/nar/gkab1107)
| [DoriC](https://tubic.org/doric/home) | oriC | v12.1 || _Nucleic Acids Res_ | 2023.1 | [DoriC 12.0: an updated database of replication origins in both complete and draft prokaryotic genomes.](https://doi.org/10.1093/nar/gkac964)
| [TIGRAFMs](http://tigrfams.jcvi.org/cgi-bin/index.cgi) | protein sequence | v15.0 | 2014.9 | _Nucleic Acids Res_ | 2001.1 | [TIGRFAMs: a protein family resource for the functional identification of proteins.](https://doi.org/10.1093/nar/29.1.41)
| [Pfam](https://www.ebi.ac.uk/interpro/download/Pfam/) | protein sequences | v37.0 | 2024.5 | _Nucleic Acids Res_ | 2021.1 | [Pfam: The protein families database in 2021.](https://doi.org/10.1093/nar/gkaa913)
| [Rfam](https://rfam.org) | microRNA | v15 | 2024.9 | _Nucleic Acids Res_ | 2021.1 | [Rfam 14: expanded coverage of metagenomic, viral and microRNA families](https://doi.org/10.1093/nar/gkaa1047)
| [PHI](https://phi5.phi-base.org) | Pathogen Host Interactions | v5.0 | 2024.3 | _Nucleic Acids Res_ | 2022.1 | [PHI-base in 2022: a multi-species phenotype database for Pathogen–Host Interactions](https://doi.org/10.1093/nar/gkab1037)
| [oriTDB](https://bioinfo-mml.sjtu.edu.cn/oriTDB2/) |||| _Nucleic Acids Res_ | 2024.10 | [oriTDB: a database of the origin-of-transfer regions of bacterial mobile genetic elements](https://doi.org/10.1093/nar/gkae869)
| [COG](https://www.ncbi.nlm.nih.gov/research/COG) |||| _Nucleic Acids Res_ | 2024.11 | [COG database update 2024](https://doi.org/10.1093/nar/gkae983) | [FTP](https://ftp.ncbi.nlm.nih.gov/pub/COG/)
| [OrthoDB](https://www.orthodb.org) || v11.0 || _Nucleic Acids Res_ | 2022.11 | [OrthoDB v11: annotation of orthologs in the widest sampling of organismal diversity](https://doi.org/10.1093/nar/gkac998)
| [CAZyme3D](https://pro.unl.edu/CAZyme3D/) | web ||| bioRxiv | 2024.12 | [CAZyme3D: a database of 3D structures for carbohydrate-active enzymes](https://doi.org/10.1101/2024.12.27.630555)
| [S9BactDB](http://caps.ncbs.res.in/S9BactDB) | S9 proteases | web || bioRxiv | 2025.1 | [S9BactDB: A database for S9 family of proteases in bacterial genomes](https://doi.org/10.1101/2025.01.01.631042)
| [FungAMR](https://github.com/Landrylab/FungAMR) | Fungi AMR ||| bioRxiv | 2024.10 | [FungAMR: A comprehensive portrait of antimicrobial resistance mutations in fungi](https://www.biorxiv.org/content/10.1101/2024.10.07.617009v1) | work with ChroQueTas

### gennome content analysis
| Tool/workflows | Language | Lastest version | Release date | Journal | Pub Date | Paper title | Notes |
| -------------- | -------- | --------------- | ------------ | ------- | -------- | ----------- | ----- |
| [doubletrouble](https://github.com/almeidasilvaf/doubletrouble) | R | v1.6.0 | 2025.2 | _Bioinformatics_ | 2025.2 | [doubletrouble: an R/Bioconductor package for the identification, classification, and analysis of gene and genome duplications](https://doi.org/10.1093/bioinformatics/btaf043)
| [gyoza](https://github.com/durr1602/gyoza) | snakemake | v11.0.0 | 2025.2 | bioRxiv | 2025.2 | [gyōza: a Snakemake workflow for modular analysis of deep-mutational scanning data](https://doi.org/10.1101/2025.02.19.639168)

### protein structure
| Tool/workflows | Language | Lastest version | Release date | Journal | Pub Date | Paper title | Notes |
| -------------- | -------- | --------------- | ------------ | ------- | -------- | ----------- | ----- |
| [Chai-1](https://github.com/chaidiscovery/chai-lab) | Python | v0.0.1 | 2024.09 |  blog | 2024.09 | [Introducing Chai-1: Decoding the molecular interactions of life](https://www.chaidiscovery.com/blog/introducing-chai-1)
| [AlphaFold2](https://github.com/google-deepmind/alphafold) | Python | v2.3.2 | 2023.04 | _Nature_ | 2021.07 | [Highly accurate protein structure prediction with AlphaFold](https://doi.org/10.1038/s41586-021-03819-2)
| [AlphaFold3](https://www.alphafoldserver.com/) | Web Service ||| _Nature_ | 2024.05 | [Accurate structure prediction of biomolecular interactions with AlphaFold 3.](https://doi.org/10.1038/s41586-024-07487-w)
| [DNAproDB](https://dnaprodb.usc.edu/) | Online database ||| _Nucleic Acids Res_ | 2024.11 | [DNAproDB: an updated database for the automated and interactive analysis of protein–DNA complexes](https://doi.org/10.1093/nar/gkae970)


### elements 
| Tool/workflows | Language | Lastest version | Release date | Journal | Pub Date | Paper title | Notes |
| -------------- | -------- | --------------- | ------------ | ------- | -------- | ----------- | ----- |
| [DeepInverton](https://github.com/HUST-NingKang-Lab/DeepInverton) | Python | NoReleaseTag || _Nucleic Acids Res_ | 2024.10 | [Deep learning revealed the distribution and evolution patterns for invertible promoters across bacterial lineages](https://doi.org/10.1093/nar/gkae966)

### bacteriophage
| Tool/workflows | Language | Lastest version | Release date | Journal | Pub Date | Paper title | Notes |
| -------------- | -------- | --------------- | ------------ | ------- | -------- | ----------- | ----- |
| [SatelliteFinder](https://hub.docker.com/r/gempasteur/satellite_finder) | Docker ||| _Nucleic Acids Res_ | 2023.3 |[Identification and characterization of thousands of bacteriophage satellites across bacteria](https://doi.org/10.1093/nar/gkad123) | [Galaxy Service ](https://galaxy.pasteur.fr/root?tool_id=toolshed.pasteur.fr/repos/fmareuil/satellitefinder/SatelliteFinder/0.9)
| [Jaeger](https://github.com/MGXlab/Jaeger) | Python | v1.3.30-alpha | 2024.8(2024.4) | bioRxiv | 2024.9 | [Jaeger: an accurate and fast deep-learning tool to detect bacteriophage sequences.](https://doi.org/10.1101/2024.09.24.612722)
| [PhageGE](http://jason-zhao.shinyapps.io/PhageGE_Update/) | Web Platform ||| _GigaScience_ | 2024.9 | [PhageGE: an interactive web platform for exploratory analysis and visualization of bacteriophage genomes](https://doi.org/10.1093/gigascience/giae074)